src/model/createModel.ts

```diff
@@ -64,14 +69,18 @@ export function createModel<
         return properties
       }
 
-      if (
-        typeof initialValue === 'string' ||
-        typeof initialValue === 'number' ||
-        typeof initialValue === 'boolean' ||
-        // @ts-ignore
-        initialValue?.constructor.name === 'Date' ||
-        Array.isArray(initialValue)
-      ) {
+      if (propertyDefinition instanceof NullableProperty) {
```

Comment:
Why are we checking for nullable properties first? Would some of them fall under `isModelValueType` check below?

Reply:
I don't think it would be caught by the `isModelValueType` check, I put it there because it felt happy next to the `instanceof PrimaryKey` check haha. I can move it below `isModelValueType` though if you'd prefer?